libslide/src/bignum/bignum_add.rs

```diff
@@ -0,0 +1,143 @@
+mod bignum_add {
+    use crate::bignum::Bignum;
+    use std::ops;
+
+    // @todo - figure out a swap method to handle lines 17-26
+    impl ops::Add for Bignum {
+        type Output = Bignum;
+
+        fn add(self, rhs: Bignum) -> Bignum {
+            // this assumes the signs are both positive for now
+            let mut carry: u8 = 0;
+            let mut lhs_size: usize = self.dec.iter().count();
+            let mut rhs_size: usize = rhs.dec.iter().count();
+            let mut res_dec: Vec<u8> = Vec::new();
+            let mut big_vec: Vec<u8>;
```

Comment:
Just make one vector to hold the resulting decimal value. Initially make it the bigger decimal, and you can allocate more if needed at the end.

Reply:
wait what? Big_Vec is need to store the larger vector of lhs and rhs?